helper/schema/resource_data.go

```diff
@@ -604,6 +607,67 @@ func (d *ResourceData) GetRawConfig() cty.Value {
 	return cty.NullVal(schemaMap(d.schema).CoreConfigSchema().ImpliedType())
 }
 
+// GetRawConfigAt is a helper method for retrieving specific values
+// from the RawConfig returned from GetRawConfig. It returns the cty.Value
+// for a given cty.Path or an error diagnostic if the value at the given path does not exist.
+//
+// GetRawConfigAt is considered advanced functionality, and
+// familiarity with the Terraform protocol is suggested when using it.
+func (d *ResourceData) GetRawConfigAt(valPath cty.Path) (cty.Value, diag.Diagnostics) {
+	rawConfig := d.GetRawConfig()
+	configVal := cty.NullVal(cty.EmptyObject)
```

Comment:
I wonder what cty typically uses as the default "we don't know what the type is" value.

I'd imagine it's something like an unknown dynamic pseudo type

Reply:
It seems to be [DynamicVal](https://github.com/zclconf/go-cty/blob/main/cty/unknown.go#L83), I've updated the default value to that which should also fix the false positive "Invalid config path" error thrown above.